pkg/pillar/base/execwrapper.go

```diff
@@ -34,39 +32,18 @@ type Command struct {
 
 // Output runs the command and returns its standard output.
 // Any returned error will usually be of type *ExitError.
-// Waits for the exec call to finish for `maxTimeOutLimit` after which timeout error is returned
+// Waits for the exec call to finish for `timeOutLimit` after which timeout error is returned
 func (c *Command) Output() ([]byte, error) {
 	var buf bytes.Buffer
 	c.command.Stdout = &buf
 	c.buffer = &buf
-	c.timeout = maxTimeOutLimit
+	c.timeout = timeOutLimit
```

Comment:
Instead of `CombinedOutputWithCustomTimeout` and `OutputWithCustomTimeout`, maybe we could add `Command.WithTimeout()` just like there is `Command.WithContext()` (or `Command.WithCustomTimeout()` to make it clear that there is some timeout either way). WDYT?
But in any case I think that 100 seconds default timeout is reasonable.

Reply:
I agree with @milan-zededa , `Command.WithCustomTimeout()` would allow any custom timeout (provided as argument), but it would need to check the watchdog timeout to not allow a timer higher than the watchdog... 100 seconds sounds reasonable, but it also depends of the kind of application being executed... @christoph-zededa , it would be good to include the watchdog timeout value in the commit message to justify the 100 seconds, you just said 1000 is higher than the watchdog, but how much higher?

Reply:
@rene what should it do if the timeout is higher than the watchdog timeout? I cannot return an error here.

Reply:
@christoph-zededa log + use the default timeout might make sense.

Reply:
@eriknordmark I now return an error, because that indicates the (future) developer that something is going wrong - I fear that a log line just gets lost and ignored.

Reply:
OK